Tests/TestingTests/EventTests.swift

```diff
@@ -0,0 +1,68 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+import Foundation
+@testable
+@_spi(ExperimentalEventHandling)
+@_spi(ExperimentalParameterizedTesting)
+@_spi(ExperimentalSnapshotting)
+@_spi(ExperimentalTestRunning)
+import Testing
+@_implementationOnly 
+import TestingInternals
+
+struct EventTests {
+  @Test("Codable",
+        arguments: [
+          Event.Kind.expectationChecked(
+            Expectation(sourceCode: SourceCode(kind: .syntaxNode("SyntaxNode")),
+                        mismatchedErrorDescription: "Mismatched Error Description",
+                        expandedExpressionDescription: "Expanded Expression Description",
+                        differenceDescription: "Difference Description",
+                        isPassing: false,
+                        isRequired: true,
+                        sourceLocation: SourceLocation()
+                       )
+          ),
+          Event.Kind.testSkipped(
+            SkipInfo(comment: "Comment",
+                     sourceContext: SourceContext(backtrace: Backtrace.current(),
+                                                  sourceLocation: SourceLocation()
+                                                 )
+                    )
+          ),
+          Event.Kind.issueRecorded(
+            Issue(kind: .system,
+                  comments: ["Comment"],
+                  sourceContext: SourceContext(backtrace: nil,
+                                               sourceLocation: nil)
+                 )
+          ),
+          Event.Kind.runStarted,
+          Event.Kind.runEnded,
+          Event.Kind.testCaseStarted,
+          Event.Kind.testCaseEnded,
+          Event.Kind.testStarted,
+          Event.Kind.testEnded,
+        ]
+  )
+
+  func codable(kind: Event.Kind) async throws {
+    let testID = Test.ID(moduleName: "ModuleName",
+                         nameComponents: ["NameComponent1", "NameComponent2"],
+                         sourceLocation: SourceLocation())
+    let event = Event(kind, testID: testID, instant: .now)
+    let eventSnapshot = Event.Snapshot(event: event)
+    let encoded = try JSONEncoder().encode(eventSnapshot)
+    let decoded = try JSONDecoder().decode(Event.Snapshot.self, from: encoded)
+
+    #expect(String(describing: decoded) == String(describing: eventSnapshot))
```

Comment:
Are you using `String(describing:)` because `Event.Snapshot` isn't equatable? Is that something we'd want to add (not necessarily for this PR)?

Reply:
Yes that's why :)